src/java/picard/illumina/IlluminaBasecallsConverter.java

```diff
@@ -453,8 +453,8 @@ public synchronized void addRecord(final String barcode, final CLUSTER_OUTPUT_RE
 
         private synchronized SortingCollection<CLUSTER_OUTPUT_RECORD> newSortingCollection() {
             final int maxRecordsInRam =
-                    maxReadsInRamPerTile /
-                            barcodeRecordWriterMap.size();
+                    Math.max(1, maxReadsInRamPerTile /
```

Comment:
I know you're just trying to protect yourself from 0 here, but 1 is a very weird number to assign to maxRecordsInRam.  Could it be e.g. 1000?